src/Message/CIMCreateCardRequest.php

```diff
@@ -97,7 +97,7 @@ protected function addBillingData(\SimpleXMLElement $data)
             $req->state = $card->getBillingState();
             $req->zip = $card->getBillingPostcode();
             $req->country = $card->getBillingCountry();
-            $req->phoneNumber = $card->getBillingPhone();
+            $req->phoneNumber = substr(trim($card->getBillingPhone()), 0, 25);
```

Comment:
let's just change this in the AuthorizeNetCard in `getDataForPersist` 
let's check validation limits for all the fields and truncate them if needed. we can append a `'…'` in the end when truncating

Refer https://developer.authorize.net/api/reference/index.html#customer-profiles-create-customer-payment-profile (Request Field Description)